compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -181,6 +184,38 @@ envInsertAlias v x env = env{envAliases = MS.insert v x (envAliases env)}
 envLookupAlias :: Var -> Env -> Maybe LF.Expr
 envLookupAlias x = MS.lookup x . envAliases
 
+-- | Bind a type var without shadowing its LF name.
+envBindTypeVar :: Var -> Env -> (TypeVarName, Env)
+envBindTypeVar x env = try 1 (TypeVarName prefix)
+    where
+        prefix = getOccText x
+        nameFor i = TypeVarName (prefix <> T.pack (show i))
+
+        try :: Integer -> TypeVarName -> (TypeVarName, Env)
```

Comment:
```suggestion
        try :: Int -> TypeVarName -> (TypeVarName, Env)
```
If you get over 2^63-1 type variable collisions you probably have bigger problems :slightly_smiling_face:

Reply:
👍